### PR TITLE
LibWeb: Add ReadableStreamGetReaderOptions to ReadableStream.getReader

### PR DESCRIPTION
**Pull request by shannonbooth**

### CODE REVIEW

File: Userland/Libraries/LibWeb/Streams/AbstractOperations.cpp [hunk 1]
```diff
@@ -481,6 +481,26 @@ WebIDL::ExceptionOr<void> set_up_readable_stream_default_reader(ReadableStreamDe
...
 
+// https://streams.spec.whatwg.org/#set-up-readable-stream-byob-reader
+WebIDL::ExceptionOr<void> set_up_readable_stream_byob_reader(ReadableStreamBYOBReader& reader, ReadableStream& stream)
+{
+    // 1. If ! IsReadableStreamLocked(stream) is true, throw a TypeError exception.
+    if (is_readable_stream_locked(stream))
+        return WebIDL::SimpleException { WebIDL::SimpleExceptionType::TypeError, "Cannot create stream reader for a locked stream"sv };
+
+    // 2. If stream.[[controller]] does not implement ReadableByteStreamController, throw a TypeError exception.
+    if (stream.controller()->has<JS::NonnullGCPtr<ReadableByteStreamController>>())
```
Comment (mattco98) [inline Userland/Libraries/LibWeb/Streams/AbstractOperations.cpp:507]: `if (!...)`

**Review by mattco98 — CHANGES_REQUESTED** [review]
Awesome! Thanks for doing this :smile:

File: Userland/Libraries/LibWeb/Streams/AbstractOperations.cpp (hunk 1 above)
Comment (shannonbooth) [reply Userland/Libraries/LibWeb/Streams/AbstractOperations.cpp]: :scream:

**Review by mattco98 — APPROVED** [review]

File: Userland/Libraries/LibWeb/Streams/ReadableStream.cpp [hunk 2]
```diff
@@ -93,14 +93,17 @@ WebIDL::ExceptionOr<JS::GCPtr<JS::Object>> ReadableStream::cancel(JS::Value reas
 }
 
 // https://streams.spec.whatwg.org/#rs-get-reader
-WebIDL::ExceptionOr<ReadableStreamReader> ReadableStream::get_reader()
+WebIDL::ExceptionOr<ReadableStreamReader> ReadableStream::get_reader(ReadableStreamGetReaderOptions options)
```
Comment (linusg) [inline Userland/Libraries/LibWeb/Streams/ReadableStream.cpp:96]: This might be a lightweight struct, but no reason to copy it - pass by const ref

**Review by linusg — CHANGES_REQUESTED** [review]
> LibWeb: Implement Streams::SetUpReadableStreamBYOBReader AO

Don't use C++ method syntax to refer to methods that don't exist - reword this as `LibWeb/Streams: Implement the SetUpReadableStreamBYOBReader AO` or something like that

File: Userland/Libraries/LibWeb/Streams/ReadableStream.cpp (hunk 2 above)
Comment (shannonbooth) [reply Userland/Libraries/LibWeb/Streams/ReadableStream.cpp]: Thanks @linusg , applied both comments :)
